src/lobby.js

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import mj from "minijanus";
+
+import Config from "./config";
+
+import "./lobby.css";
+
+class Lobby extends React.Component {
+  constructor() {
+    super();
+    this.state = { rooms: [] };
+
+    this.onWebsocketOpen = this.onWebsocketOpen.bind(this);
+    this.onWebsocketMessage = this.onWebsocketMessage.bind(this);
+    this.updateRooms = this.updateRooms.bind(this);
+  }
+
+  componentDidMount() {
+    this.ws = new WebSocket(Config.janus_server_url, "janus-protocol");
+    this.session = new mj.JanusSession(this.ws.send.bind(this.ws));
+    this.ws.addEventListener("open", this.onWebsocketOpen);
+    this.ws.addEventListener("message", this.onWebsocketMessage);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.pollRoomsIntervalHandle);
+    this.ws.removeEventListener("open", this.onWebsocketOpen);
+    this.ws.removeEventListener("message", this.onWebsocketMessage);
+  }
+
+  onWebsocketOpen() {
+    this.session
+      .create()
+      .then(() => {
+        this.handle = new mj.JanusPluginHandle(this.session);
+        return this.handle.attach("janus.plugin.sfu").then(this.updateRooms);
+      })
+      .then(() => {
+        this.pollRoomsIntervalHandle = setInterval(this.updateRooms, 10000);
+      });
```

Comment:
Maybe fine for now but we should probably just adopt `async`/`await` syntax instead of `.then(...)`. It makes for much nicer code.

Reply:
Yeah agree, I didn't feel like figuring out how to set up the tooling for that. Robert already did it for the janus adapter so I figured I would just let him handle it :)

Reply:
Babel is already set up to transform async/await with `babel-preset-env`. We can change it whenever.